src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultObjectValidator.cs

```diff
@@ -51,14 +54,14 @@ public class DefaultObjectValidator : IObjectModelValidator
                 throw new ArgumentNullException(nameof(actionContext));
             }
 
-            if (validatorProvider == null)
+            if (validationState == null)
```

Comment:
The `ValidationVisitor` handles `validationState == null` just fine. Why does this condition now lead to an `ArgumentNullException`?